test/src/tests-persistence.ts

```diff
@@ -1877,6 +1877,25 @@ describe('migrations and persistence-related', () => {
         done();
     });
 
+    it('should prioritize device id set via mParticle.config instead of local storage', done => {
+        mParticle._resetForTests(MPConfig);
+
+        setLocalStorage();
+
```

Comment:
Can you do a check for the deviceId at this point also to ensure there is a value here?

Reply:
To clarify - mParticle.init in order to get a real guid device id. check for that to exist. then reinitialize with a device id on the config.